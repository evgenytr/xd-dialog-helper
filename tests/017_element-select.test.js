const select = require('../src/main').types.SELECT;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {}),
    close: jest.fn(() => {}),
    cancel: jest.fn(() => {}),
    registerElement: jest.fn(() => undefined)
};

const rendered = select.render('dialog', {
    id: 'select',
    label: 'Move right',
    value: 'a',
    options: [
        {
            value: 'a',
            label: 'A'
        },
        {
            value: 'b',
            label: 'B'
        }
    ],
    htmlAttributes: {
        someAttr: 'hello'
    }
}, actions);

global.console.error = jest.fn(() => {});

describe('Select Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();

            expect(select.render('dialog', {id: 'noHTML', options: []}, actions)).toMatchSnapshot();
        });

        it('should log an error to the console for missing configuration', () => {
            expect(select.render('dialog', {
                id: 'select',
                label: 'Hello'
            }, actions)).toBeNull();
            expect(global.console.error).toBeCalled();
        });
    });
    describe('value()', () => {
        it('should return the correct value', () => {
            expect(select.value(rendered)).toBe('a');

        });

        it('should correctly handle changing values', () => {
            rendered.input.value = 'b';
            rendered.input.dispatchEvent(new Event('change'));

            expect(actions.change).toBeCalled();
            expect(select.value(rendered)).toBe('b');
        });
    });
    describe('validate()', () => {
        it('should always return true', () => {
            expect(select.valid(rendered)).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(select.type).toMatchSnapshot();
    });
});
