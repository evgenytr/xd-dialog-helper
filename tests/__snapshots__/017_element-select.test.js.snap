// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Select"`;

exports[`Select Element render() should render correctly 1`] = `
Object {
  "input": <select
    id="dialog-select"
    someattr="hello"
  >
    <option
      value="a"
    >
      A
    </option>
    <option
      value="b"
    >
      B
    </option>
  </select>,
  "props": Object {
    "htmlAttributes": Object {
      "someAttr": "hello",
    },
    "id": "select",
    "label": "Move right",
    "options": Array [
      Object {
        "label": "A",
        "value": "a",
      },
      Object {
        "label": "B",
        "value": "b",
      },
    ],
    "value": "a",
  },
  "type": Object {
    "render": [Function],
    "type": "Select",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <label
    id="dialog-select-wrapper"
  >
    <span
      id="dialog-select-label"
    >
      Move right
    </span>
    <select
      id="dialog-select"
      someattr="hello"
    >
      <option
        value="a"
      >
        A
      </option>
      <option
        value="b"
      >
        B
      </option>
    </select>
  </label>,
}
`;

exports[`Select Element render() should render correctly 2`] = `
Object {
  "input": <select
    id="dialog-noHTML"
  />,
  "props": Object {
    "id": "noHTML",
    "options": Array [],
  },
  "type": Object {
    "render": [Function],
    "type": "Select",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <label
    id="dialog-noHTML-wrapper"
  >
    <span
      id="dialog-noHTML-label"
    >
      undefined
    </span>
    <select
      id="dialog-noHTML"
    />
  </label>,
}
`;
